### DISCUSSION TITLE
Constant logging of REFRESH_TOKEN_ERROR

### DISCUSSION THREAD

**Discussion by Eldainorn** [keycloak/keycloak#19798] [Q&A]
Hey everyone
Asking for help regarding some issue with Keycloak. I will try to give all relevant information and hopefully anyone has an idea what might the issue be.
We have a prod environment with Keycloak v9 (will go to v20 in a few months) and it has been in use for several years now. For some reason the logs containting the following WARN message have started pilling up immensely.
`type=REFRESH_TOKEN_ERROR, realmId=Standalone902, clientId=agg, userId=null, ipAddress=SOME_IP_I_REMOVED_IT, error=invalid_token, grant_type=refresh_token, client_auth_method=client-secret`

One would think that the answer is self explanatory. Refresh token is not valid anymore. But looking at dev console in Chrome, there are no issues popping up in it, login is working perfectly as well as navigating the application. Even request with refresh token are working perfectly OK.
Looking at Keycloak admin console settings, SSO session idle is set to 30min, Access Token Lifespan is set to 5 minutes. Which also seems good.

Should these log messages be treated as false positives? We have no idea what might have started causing them. But it is annoying since log file is growing so fast in size.

**Comment by sschu** [comment]
Have you verified where that request is coming from i.e. is the source IP the one you expect requests coming from?

**Comment by Eldainorn** [comment]
Sorry for late response, I was on vacation. 
Yes, the sys guys confirmed that IP the requests are coming from is our internal. That is why we are confused where this could be coming from, per my question above that checking the client dev console, no visible requests are present.

**Reply by sschu** [reply]
If it is from your network, couldn't you identify the exact source?

**Comment by stevejr** [comment]
@Eldainorn - did you find the resolution to this issue?
